extension/chrome/elements/compose-modules/compose-send-btn-module.ts

```diff
@@ -194,65 +200,63 @@ export class ComposeSendBtnModule extends ViewModule<ComposeView> {
     return { mimeType, data };
   };
 
-
-  private doSendMsg = async (msg: SendableMsg) => {
+  private doSendMsgs = async (msgObj: MultipleMessages): Promise<SendMsgsResult> => {
     // if this is a password-encrypted message, then we've already shown progress for uploading to backend
     // and this requests represents second half of uploadable effort. Else this represents all (no previous heavy requests)
     const progressRepresents = this.view.pwdOrPubkeyContainerModule.isVisible() ? 'SECOND-HALF' : 'EVERYTHING';
-    let msgSentRes: GmailRes.GmailMsgSend;
-    try {
-      this.isSendMessageInProgress = true;
-      msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
-    } catch (e) {
-      if (msg.thread && ApiErr.isNotFound(e) && this.view.threadId) { // cannot send msg because threadId not found - eg user since deleted it
-        msg.thread = undefined;
-        msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
-      } else {
-        this.isSendMessageInProgress = false;
-        throw e;
-      }
+    const sentIds: string[] = [];
+    const operations = [this.view.draftModule.draftDelete()]; // todo: don't delete on error
```

Comment:
I think we only should start `draftDelete` operation after all sends are successful?

Reply:
ping @tomholub

Reply:
I think so